MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -46,34 +46,99 @@ public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto reg
         var user = mapper.Map<UserEntity>(registrationDto);
 
         user.Password = BCrypt.HashPassword(user.Password);
+        await context.Users.AddAsync(user);
 
-        context.Users.Add(user);
+        var tokenPair = authenticationManager.IssueTokenPair(user);
+
+        await context.RefreshTokens.AddAsync(tokenPair.RefreshToken);
         await context.SaveChangesAsync();
 
-        var accessToken = authenticationManager.IssueAccessToken(user);
         var userInfoDto = mapper.Map<UserOutputDto>(user);
         var outputDto = new UserRegistrationResultDto
         {
             UserInfo = userInfoDto,
-            AccessToken = accessToken
+            TokenPair = new() { AccessToken = tokenPair.AccessToken, RefreshTokenId = tokenPair.RefreshToken.Id }
         };
         return CreatedAtAction(nameof(GetUserById), new {id = outputDto.UserInfo.Id}, outputDto);
     }
 
+    [HttpPost("refreshTokens/{id:guid}")]
+    public async Task<IActionResult> TokenRefresh(Guid id)
+    {
+        var refreshToken = await context.RefreshTokens.Where(refreshToken => refreshToken.Id == id).SingleOrDefaultAsync();
+
+        if(refreshToken is null)
+        {
+            return BadRequest($"Token with id: {id} doesn't exist.");
+        }
+
+        var user = await context.Users.Where(user => user.Id == refreshToken.UserId).SingleOrDefaultAsync();
+
+        var tokenPair = authenticationManager.IssueTokenPair(user);
+
+        context.RefreshTokens.Remove(refreshToken);
+        await context.RefreshTokens.AddAsync(tokenPair.RefreshToken);
+        await context.SaveChangesAsync();
+
+        var outputDto = new AuthenticationTokenPairOutputDto { AccessToken = tokenPair.AccessToken, RefreshTokenId = tokenPair.RefreshToken.Id };
+        return Ok(outputDto);
+    }
+
+    [HttpDelete("{userId:guid}/refreshTokens")]
```

Comment:
This action should require user to be authorized, or any user will be able to kick out another user from their session.